libraries/chain/controller.cpp

```diff
@@ -2592,20 +2592,51 @@ block_state_ptr controller::finalize_block( const signer_callback_type& signer_c
 
    auto& ab = std::get<assembled_block>(my->pending->_block_stage);
 
+   auto pfa = *ab._pending_block_header_state.prev_activated_protocol_features.get();
```

Comment:
`prev_activated_protocol_features` is a shared_ptr, why not just pass the shared_ptr around?

Reply:
Done.